route/pool.go

```diff
@@ -473,6 +484,25 @@ func (p *EndpointPool) MarshalJSON() ([]byte, error) {
 	return json.Marshal(endpoints)
 }
 
+// OverrulePoolLoadBalancingAlgorithm Overwrites the load balancing algorithm of a pool by that of a specified endpoint, if that is valid.
+func (p *EndpointPool) OverrulePoolLoadBalancingAlgorithm(endpoint *Endpoint) {
+	p.Lock()
+	defer p.Unlock()
+	if len(endpoint.LoadBalancingAlgorithm) > 0 && endpoint.LoadBalancingAlgorithm != p.LoadBalancingAlgorithm {
+		if config.IsLoadBalancingAlgorithmValid(endpoint.LoadBalancingAlgorithm) {
+			// Multiple apps can have the same route, a pool will get the last endpoint's algorithm
```

Comment:
What should this comment tell the reader? The fact that a single route can be added to many apps does not interfere with the per-route options as those are attached to the route, not the app.